### DISCUSSION TITLE
Accessing highlight/locals query results

### DISCUSSION THREAD

**Discussion by tinygiant98** [tree-sitter/tree-sitter#1554] [Q&A]
I'm finally back at it, trying to integrate a tree-sitter parser into vscode a for custom language, and it seems to work.  I've recently started experimenting with queries to assign captures to specific nodes.  The highlighting works great.  My current extension solution directly accesses the tree nodes to obtain the node_type or value.  It seems to work well.  My question is, after having created and run the highlighting and locals queries, do they become part of the tree, or can they become part of the tree, and if so, would they be accessible much like the nodes are?  If they aren't part of the tree, can the query results be accessed by another method?

**Comment by tinygiant98** [comment]
I've put quite a bit of time into trying to pull query data out of web-tree-sitter and have had some luck with it.  The query data is not exported, so it's buried behind some deep syntax, like `tree.getLanguage().query(query).matches(node);`  Based on the function prototype, web-tree-sitter only accepts a single `SyntaxNode` to conduct the query against, which mean in order to check for specific cases in an entire document, I have to re-run the query against each traversed node.  The highlight/query functionality isn't exactly quick to begin with, so I'd like to achieve a bit more efficiency here.  Also, there doesn't seem to be a way to include scopes (like with `locals.scm`) in this query methodology.

If anyone has any experience using queries through web-tree-sitter, I'd love to find out if it's possible to pass an entire tree and run the query against all nodes versus running the query several thousand times in a single document.

**Reply by ObserverOfTime** [reply]
> which mean in order to check for specific cases in an entire document, I have to re-run the query against each traversed node.

Or just run it on the root node?
